src/hotspot/share/runtime/thread.inline.hpp

```diff
@@ -204,13 +190,18 @@ inline bool JavaThread::is_terminated() const {
   return check_is_terminated(l_terminated);
 }
 
-inline void JavaThread::set_terminated(TerminatedTypes t) {
```

Comment:
I prefer set_terminated(arg) over the new set of methods.

Reply:
We had two methods:
```
   void set_terminated(TerminatedTypes t);
   void set_terminated_value();
```
Terminated is both the name of the flag, the name of the type and part of the names of two of the states, which is very confusing.

Also the setters now match the queries:
E.g.
`bool is_exiting()`

The queries do not indicate in any sense that they are queries on the terminated flag.
The state flag is an implementation detail from query POV.
So to be consistent e.g. "set_exiting()" also hides the fact that we keep track of this with a flag.

Reply:
Please advise :) , I can roll back if you insist!